bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java

```diff
@@ -0,0 +1,600 @@
+        }
+    }
+
+    /**
+     * Build request URL from configuration data
+     *
+     * @return a valid URL for the thermostat's JSON interface
+     */
+    private String buildRequestURL(String resource) {
+        RadioThermostatConfiguration config = getConfigAs(RadioThermostatConfiguration.class);
+
+        String hostName = config.hostName;
+
+        String urlStr = URL.replace("%hostName%", hostName);
+        urlStr = urlStr.replace("%resource%", resource);
+
+        return urlStr;
+    }
+
+    /**
+     * Request new data from the thermostat
+     *
+     * @param the resource URL constant for a particular thermostat JSON resource
+     * @return an object mapping to one of the various thermostat JSON responses or null in case of error
+     */
+    private @Nullable Object getRadioThermostatData(String resource) {
+        Object result = null;
+        String errorMsg = null;
+
+        String urlStr = buildRequestURL(resource);
+        logger.debug("URL = {}", urlStr);
+
+        try {
+            // Run the HTTP request and get the JSON response from the thermostat
+            ContentResponse contentResponse = httpClient.newRequest(urlStr).method(GET).timeout(20, TimeUnit.SECONDS)
+                    .send();
+
+            String response = contentResponse.getContentAsString();
+            logger.debug("thermostatResponse = {}", response);
+
+            // Map the JSON response to the correct object
+            if (DEFAULT_RESOURCE.equals(resource)) {
+                result = gson.fromJson(response, RadioThermostatJsonResponse.class);
+            } else if (HUMIDITY_RESOURCE.equals(resource)) {
+                result = gson.fromJson(response, RadioThermostatJsonHumidity.class);
+            } else if (RUNTIME_RESOURCE.equals(resource)) {
+                result = gson.fromJson(response, RadioThermostatJsonRuntime.class);
+            } else if (MODEL_RESOURCE.equals(resource)) {
+                result = gson.fromJson(response, RadioThermostatJsonModel.class);
+            } else if (NAME_RESOURCE.equals(resource)) {
+                result = gson.fromJson(response, RadioThermostatJsonName.class);
+            }
+
+            if (result != null) {
+                if (initialized == 1) {
+                    updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE,
+                            rthermData.getName() + " " + rthermData.getModel());
+                }
+                return result;
+            } else {
+                retryCounter++;
+                if (retryCounter == 1) {
+                    logger.warn("Error in contacting the thermostat, retrying once");
+                    return getRadioThermostatData(resource);
+                }
+                errorMsg = "missing data object";
+                logger.warn("Error in thermostat response: {}", errorMsg);
+            }
+
+        } catch (IllegalStateException | InterruptedException | TimeoutException | ExecutionException e) {
```

Comment:
What do you expect to throw an IllegalStateException?

Reply:
probably leftover from when I was doing regexes here. removed.